weed/storage/volume.go

```diff
@@ -23,6 +28,9 @@ type Volume struct {
 
 	dataFileAccessLock sync.Mutex
 	lastModifiedTime   uint64 //unix time in seconds
+
+	lastCompactingIndexOffset       uint64
+	incrementedHasUpdatedIndexEntry map[uint64]keyField
```

Comment:
Yeah, I know current garbage collection is ordered(one by one volume). But I have the concern that when concurrent compacting is involved in weed some day, these 2 fields should become volume specific.